### DISCUSSION TITLE
A LocalModule storage option

### DISCUSSION THREAD

**Discussion by jacksund** [PrefectHQ/prefect#3304] [Ideas] [answered]
Is there an LocalModule Storage option? I feel that this will be more convenient for testing than Github or Local storage. My thought is that LocalModule will serve just like Local with stored_as_script=True, where instead of a path='/path/to/my/flows/exampleflow.py', you could use module_path='from myrepo.flows import exampleflow'. Thus the flow is accessible in any environment where that module import works, not just the environments that share the same path structure. 

In the scientific community, this would help the sharing of flow libraries. See [Atomate](https://github.com/hackingmaterials/atomate) as an example -- this is a python library filled with pre-built Tasks and WorkFlows pertaining to materials chemistry. You can achieve the same flow sharing with Github storage, but I feel like this is more hoops to jump through and also slower in execution time.

Thoughts?

**Comment by cicdw** [comment, marked as answer]
Hi @jacksund - I think this makes sense!  I don't expect it will be a common use case, but if we can support it with minimal branches to our current code paths then I'm all for it.  I'll convert this to an issue and we can keep the conversation going about what the full implementation might look like / how feasible it ends up being.

**Reply by cicdw** [reply]
It appears I cannot convert discussions to issues - would you mind opening an issue for this?

**Reply by jacksund** [reply]
Done! It's issue #3335
